src/Text/TeXMath/Types.hs

```diff
@@ -85,10 +88,11 @@ data Exp =
   | ESqrt Exp      -- ^ A square root.
   | EScaled Rational Exp -- ^ An expression that is scaled to some factor
                   -- of its normal size.
-  | EArray [Alignment] [ArrayLine] -- ^ An array or matrix.  The first argument
+  | EArray [Alignment] [ArrayLine] [ColumnSeparator] -- ^ An array or matrix.  The first argument
                   -- specifies the alignments of the columns; the second gives
-                  -- the contents of the lines.  All of these lists should be
-                  -- the same length.
+                  -- the contents of the lines; the third gives the type of separator between
+                  -- columns. The first two of these lists should be the same length, the last
```

Comment:
This isn't quite right, is it?  Rather, each ArrayLine should be the same length as the list of alignments.

Reply:
Conceptually, I'd prefer
```
EArray [Alignment] [ColumnSeparator] [ArrayLine]
```
with the content at the end and the "form" at the beginning.